src/main/java/org/HdrHistogram/AbstractHistogram.java

```diff
@@ -729,13 +729,10 @@ public void add(final AbstractHistogram otherHistogram) throws ArrayIndexOutOfBo
      */
     public void subtract(final AbstractHistogram otherHistogram)
             throws ArrayIndexOutOfBoundsException, IllegalArgumentException {
-        long highestRecordableValue = valueFromIndex(countsArrayLength - 1);
-        if (highestRecordableValue < otherHistogram.getMaxValue()) {
-            if (!isAutoResize()) {
-                throw new ArrayIndexOutOfBoundsException(
-                        "The other histogram includes values that do not fit in this histogram's range.");
-            }
-            resize(otherHistogram.getMaxValue());
+        if (highestEquivalentValue(otherHistogram.getMaxValue()) >
+                highestEquivalentValue(valueFromIndex(this.countsArrayLength - 1))) {
```

Comment:
It makes sense to me to use the same exception for both cases; users probably won't care about the distinction.